ext/mysqli/mysqli_api.c

```diff
@@ -1181,9 +1190,15 @@ PHP_FUNCTION(mysqli_fetch_field_direct)
 		RETURN_THROWS();
 	}
 
+	if (offset < 0) {
+		zend_argument_value_error(ERROR_ARG_POS(2), "must be greater than or equal to 0");
+		RETURN_THROWS();
+	}
+
 	MYSQLI_FETCH_RESOURCE(result, MYSQL_RES *, mysql_result, "mysqli_result", MYSQLI_STATUS_VALID);
 
-	if (offset < 0 || offset >= (zend_long) mysql_num_fields(result)) {
+	if (offset >= (zend_long) mysql_num_fields(result)) {
+		// TODO ValueError?
```

Comment:
I think that should be ok to promote because you usually know how many fields you retrieve, so `mysql_num_fields()` don't't even have to be called in most of the times.

I do really hope that people don't use `mysqli_fetch_field_direct()` with a `SELECT *` query. :)